java/bench/hive/src/java/org/apache/orc/bench/hive/rowfilter/RowFilterInputState.java

```diff
@@ -0,0 +1,116 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.orc.bench.hive.rowfilter;
+
+import org.apache.commons.lang.reflect.FieldUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
+import org.apache.orc.OrcFile;
+import org.apache.orc.OrcFilterContext;
+import org.apache.orc.Reader;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.bench.core.Utilities;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+import java.io.IOException;
+import java.util.Random;
+
+@State(Scope.Thread)
+public abstract class RowFilterInputState {
+
+    private static final Path root = new Path(System.getProperty("user.dir"));
+
+    Configuration conf = new Configuration();
+    FileSystem fs;
+    TypeDescription schema;
+    VectorizedRowBatch batch;
+    Path path;
+    boolean[] include;
+    Reader reader;
+    Reader.Options readerOptions;
+    boolean[] filterValues = null;
+
+    @Setup
+    public void setup() throws IOException, IllegalAccessException {
+        TypeDescription.RowBatchVersion version =
+            (TypeDescription.RowBatchVersion)FieldUtils.readField(this, "version", true);
+        TypeDescription.Category benchType = (TypeDescription.Category)FieldUtils.readField(this, "benchType", true);
+        String filterPerc = (String)FieldUtils.readField(this, "filterPerc", true);
+        int filterColsNum = (int)FieldUtils.readField(this, "filterColsNum", true);
+        String dataRelativePath = (String)FieldUtils.readField(this, "dataRelativePath", true);
+        String schemaName = (String)FieldUtils.readField(this, "schemaName", true);
+        String filterColumn = (String)FieldUtils.readField(this, "filterColumn", true);
+
+        fs = FileSystem.getLocal(conf).getRaw();
+        path = new Path(root, dataRelativePath);
+        schema = Utilities.loadSchema(schemaName);
+        batch = schema.createRowBatch(version, 1024);
+        include = new boolean[schema.getMaximumId() + 1];
+        for(TypeDescription child: schema.getChildren()) {
+            if (schema.getFieldNames().get(child.getId()-1).compareTo(filterColumn) == 0) {
+                System.out.println("Apply Filter on column: " + schema.getFieldNames().get(child.getId()-1));
+                include[child.getId()] = true;
+            } else if (child.getCategory() == benchType) {
+                System.out.println("Skip column(s): " + schema.getFieldNames().get(child.getId()-1));
+                include[child.getId()] = true;
+                if (--filterColsNum == 0) break;
+            }
+        }
+        if (filterColsNum != 0) {
+            System.err.println("Dataset does not contain type: " + benchType);
+            System.exit(-1);
```

Comment:
findbugs prompt
```
[INFO] org.apache.orc.bench.hive.rowfilter.RowFilterInputState.setup() invokes System.exit(...), which shuts down the entire virtual machine [org.apache.orc.bench.hive.rowfilter.RowFilterInputState] At RowFilterInputState.java:[line 80] DM_EXIT
```
Also use System.exit before modifying，I don't know how to get past the findbugs check